### PR TITLE
add conflicting_shifts signup checker

### LINKED ISSUE #233: Add "conflicting shifts" signup checker to BaseSignupMethod
The base signup method should also check whether there are any conflicting shifts for that user (overlapping times)

### PR DESCRIPTION
**Pull request by jeriox**
closes #233

### CODE REVIEW

File: ephios/plugins/basesignup/templates/basesignup/section_based/fragment_participant.html [hunk 1]
```diff
@@ -17,6 +17,13 @@ <h6 class="mb-0">
                     {% endif %}
                 </div>
                 <div>
+                    {% with form.instance.participant|conflicting_shifts:shift as conflicting_shifts %}
+                        {% if conflicting_shifts %}
+                            <span class="fa fa-exclamation-triangle text-danger"
+                                  data-toggle="tooltip"
+                                  title="{{ form.instance.participant }} already participates in {{ conflicting_shifts|join:", " }} at the same time."></span>
+                        {% endif %}
+                    {% endwith %}
```
Comment (felixrindt) [inline ephios/plugins/basesignup/templates/basesignup/section_based/fragment_participant.html:26]: We still need to merge that with the section signup participant fragment...

**Review by felixrindt — CHANGES_REQUESTED** [review]

File: ephios/plugins/basesignup/templates/basesignup/section_based/fragment_participant.html (hunk 1 above)
Comment (jeriox) [reply ephios/plugins/basesignup/templates/basesignup/section_based/fragment_participant.html:26]: not sure if that is actually possible as they use a different base layout (col-8 vs col-10)

Comment (jeriox) [reply ephios/plugins/basesignup/templates/basesignup/section_based/fragment_participant.html:26]: also section_based renders the desired section below the participants name which means that just adding another column is not enough

Comment (felixrindt) [reply ephios/plugins/basesignup/templates/basesignup/section_based/fragment_participant.html:26]: ah alright.

File: ephios/core/templates/core/disposition/fragment_participation.html [hunk 2]
```diff
@@ -8,6 +8,13 @@ <h6 class="mb-0">
             <div class="col-10 pr-3">
                 {{ form.instance.participant.icon }}
                 {{ form.instance.participant }}
+                {% with form.instance.participant|conflicting_shifts:shift as conflicting_shifts %}
+                    {% if conflicting_shifts %}
+                        <span class="fa fa-exclamation-triangle text-danger"
+                              data-toggle="tooltip"
+                              title="{{ form.instance.participant }} already participates in {{ conflicting_shifts|join:", " }} at the same time."></span>
+                    {% endif %}
+                {% endwith %}
```
Comment (felixrindt) [inline ephios/core/templates/core/disposition/fragment_participation.html:17]: You should probably call `handleForms` in the `disposition.js` when adding new forms to get the tooltip displayed correctly.

**Review by felixrindt — CHANGES_REQUESTED** [review]
So simple but such a nice feature!

**Review by felixrindt — APPROVED** [review]

**Pull request merged**